src/test/java/com/scottlogic/AuthorPostSorterTest.java

```diff
@@ -0,0 +1,76 @@
+package com.scottlogic;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class AuthorPostSorterTest {
+    @Test
+    void nullTest() {
+        List<UserPost> actual = new AuthorPostSorter().sort(null);
+
+        assertNull(actual);
+    }
+
+    @Test
+    void emptyListTest() {
+        List<UserPost> actual = Collections.emptyList();
+        List<UserPost> expected = Collections.emptyList();
+
+        actual = new AuthorPostSorter().sort(actual);
```

Comment:
In general you don't want side effects and mutation inside tests. Instead of setting `actual` to be `Collections.emptyList()` and then also the sorted version having a variable that is `initialList` or something similar makes the tests significantly clearer. The style of testing is different to normal coding as it is all about maximizing readability rather than efficiency as they will be run a finite number of times, this means a slight efficiency lose doesn't have much impact.

Reply:
This makes sense, and avoids accidentally incorrectly manipulating the actual